### PR TITLE
Raise if version is not provided in Docs task config

### PR DESCRIPTION
**Pull request by eksperimental**
It will raise with ArgumentError if :version key is not provided in config.

/cc @ericmj

### CODE REVIEW

File: lib/mix/tasks/docs.ex [hunk 1]
```diff
@@ -140,6 +140,14 @@ defmodule Mix.Tasks.Docs do
     end
   end
 
+  defp fetch_version!(config) do
+    Keyword.fetch!(config, :version)
```
Comment (josevalim) [inline lib/mix/tasks/docs.ex]: If we are going to raise, then we should use Keyword.fetch + pattern matching.

Comment (josevalim) [inline lib/mix/tasks/docs.ex]: I.e. we shouldn't rescue exceptions. I believe when @ericmj mentioned `Keyword.fetch!` he was not expecting a custom error message.

**Comment by ericmj** [comment]
I don't think we need a custom error message since it should always be caught by the compiler. Just using `Keyword.fetch!/2` should be fine.

File: lib/mix/tasks/docs.ex (hunk 1 above)
Comment (eksperimental) [inline lib/mix/tasks/docs.ex]: Sorry, this was really wrong. Too tired I guess :tired_face:

Comment (eksperimental) [inline lib/mix/tasks/docs.ex]: fixed

**Comment by eksperimental** [comment]
@ericmj It will be caught. I thought it was better to give a more meaningful message. But it's up to you guys

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**